src/HookFilter.ts

```diff
@@ -1,19 +1,31 @@
 export class HookFilter<T extends string> {
+    static convertIgnoredHooksToEnabledHooks<T extends string>(explicitlyEnabledHooks: T[] = [], ignoredHooks: T[], defaultHooks: T[]): T[] {
+        const resultHookSet = new Set([
+            ...explicitlyEnabledHooks,
+            ...defaultHooks,
+        ]);
+
+        // For each ignored hook, remove anything that matches.
+        for (const ignoredHook of ignoredHooks) {
+            resultHookSet.delete(ignoredHook);
+            // If the hook is a "root" hook name, remove all children.
+            for (const enabledHook of resultHookSet) {
+                if (enabledHook.startsWith(`${ignoredHook}.`)) {
+                    resultHookSet.delete(enabledHook);
+                } 
+            }
+        }
+
+        return [...resultHookSet];
+    }
+
     constructor(
-        public readonly defaultHooks: T[],
         public enabledHooks: T[] = [],
-        public ignoredHooks: T[] = []
     ) {
 
     }
 
     public shouldSkip(...hookName: T[]) {
-        if (hookName.some(name => this.ignoredHooks.includes(name))) {
-            return true;
-        }
-        if (hookName.some(name => this.enabledHooks.includes(name))) {
-            return false;
-        }
-        return !hookName.some(h => this.defaultHooks.includes(h));
+        return !hookName.some(name => this.enabledHooks.includes(name));
```

Comment:
Speaking about positive naming:
Isn't `foo.every()` the same as `!foo.some()`?
```suggestion
        return hookName.every(name => this.enabledHooks.includes(name));
```

Reply:
Aha, yes.